### PR TITLE
Fixes addErrorLog to Have 1 Param

### PR DESCRIPTION
**Pull request by seemavora**

### CODE REVIEW

File: api/routes/user.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ const {
 } = require('../constants').STATUS_CODES;
 const membershipState = require('../constants').MEMBERSHIP_STATE;
 
+
```
Comment (hsorathia) [inline api/routes/user.js:26]: remove this space

File: api/routes/user.js [hunk 2]
```diff
@@ -339,6 +345,7 @@ router.post('/getPagesPrintedCount', (req, res) => {
   }
   User.findOne({ email: req.body.email }, function(error, result) {
     if (error) {
+      // addErrorLog(userEmail, errorTime, apiEndpoint, errorDescription);
```
Comment (hsorathia) [inline api/routes/user.js:348]: remove comments

File: api/routes/user.js [hunk 3]
```diff
@@ -183,6 +184,7 @@ router.post('/delete', (req, res) => {
 
   User.deleteOne({ email: req.body.email }, function(error, user) {
     if (error) {
+      // addErrorLog(null, null, 'user/delete', 'error');
```
Comment (ThaiQ) [inline api/routes/user.js:187]: Remove these comments

File: api/routes/ErrorLog.js [hunk 4]
```diff
@@ -5,8 +5,7 @@ const { OK, BAD_REQUEST } = require('../constants').STATUS_CODES;
 const { addErrorLog } = require('../util/errorLog');
 
 router.post('/addErrorLog', async (req, res) => {
-  if (!await addErrorLog(req.body.userEmail, req.body.errorTime,
-    req.body.apiEndpoint, req.body.errorDescription)) {
+  if (!await addErrorLog(req)) {
```
Comment (ThaiQ) [inline api/routes/ErrorLog.js:8]: Instead if req, you could send in req.body so it can be more clean

**Comment by evanugarte** [comment]
@seemavora fix comments

File: api/routes/ErrorLog.js (hunk 4 above)
Comment (seemavora) [reply api/routes/ErrorLog.js:8]: If I do that I get an error. That was one of the first things I tried.

Comment (evanugarte) [reply api/routes/ErrorLog.js:8]: You have to figure out why it is giving you an error.

File: api/util/errorLog.js [hunk 5]
```diff
@@ -9,13 +9,13 @@ const ErrorLog = require('../models/ErrorLog');
...
  */
-async function addErrorLog(userEmail, errorTime, apiEndpoint, errorDescription){
+async function addErrorLog(req){
   let errorSaved = true;
   const newError = new ErrorLog({
-    userEmail,
-    errorTime,
-    apiEndpoint,
-    errorDescription
+    userEmail: req.body.userEmail,
```
Comment (evanugarte) [inline api/util/errorLog.js:15]: if you change the parameter thai said, what do you have to change here?

File: api/routes/Advertisement.js [hunk 6]
```diff
@@ -12,6 +13,13 @@ router.post('/addAdvertisement', (req, res) => {
 
   newAd.save(function(error) {
     if (error) {
+      // const info = {
+      //   userEmail: '',
+      //   errorTime: new Date(),
+      //   apiEndpoint: 'Advertisement/addAdvertisement',
+      //   errorDescription: 'error'
+      // };
+      // addErrorLog(info);
```
Comment (hsorathia) [inline api/routes/Advertisement.js:22]: Get rid of the comments

File: api/routes/officerManager.js [hunk 7]
```diff
@@ -72,6 +80,13 @@ router.post('/delete', (req, res) => {
   Manager.deleteOne({ email: req.body.email }, function(error, form) {
     if (error) {
       logger.log(`Officer Manager /Delete3DForm error: ${error}`);
+      const info = {
+        userEmail:req.body.email,
```
Comment (hsorathia) [inline api/routes/officerManager.js:84]: put a space in between the colon and req.body so it looks like
`userEmail: req.body.email,`

File: api/routes/3DPrintingForm.js [hunk 8]
```diff
@@ -30,6 +31,13 @@ router.post('/submit', (req, res) => {
   };
   PrintingForm3D.create(data, (error, post) => {
     if (error) {
+      // const info = {
+      //   userEmail: req.body.email,
+      //   errorTime: new Date(),
+      //   apiEndpoint: '3DPrintingForm/submit',
+      //   errorDescription: 'error'
+      // };
+      // addErrorLog(info);
```
Comment (law-nguyen) [inline api/routes/3DPrintingForm.js:40]: no comments pls :">

Comment (seemavora) [reply api/routes/3DPrintingForm.js:40]: oops my bad thought i removed them!

File: api/routes/3DPrintingForm.js [hunk 9]
```diff
@@ -45,6 +46,13 @@ router.post('/GetForm', (req, res) => {
...
   PrintingForm3D.find(obj, (error, forms) => {
     if (error) {
+      const info = {
+        userEmail: req.body.email,
+        errorTime: new Date(),
+        apiEndpoint: '3DPrintingForm/GetForm',
+        errorDescription: 'error'
+      };
+      addErrorLog(info);
       logger.log(`3DPrinting /GetForm error: ${error}`);
```
Comment (ThaiQ) [inline api/routes/3DPrintingForm.js:56]: @evanugarte  @seemavora  Do we still need these loggers?

File: api/routes/3DPrintingForm.js [hunk 10]
```diff
@@ -45,6 +46,13 @@ router.post('/GetForm', (req, res) => {
 
   PrintingForm3D.find(obj, (error, forms) => {
     if (error) {
+      const info = {
+        userEmail: req.body.email,
+        errorTime: new Date(),
+        apiEndpoint: '3DPrintingForm/GetForm',
+        errorDescription: 'error'
```
Comment (ThaiQ) [inline api/routes/3DPrintingForm.js:53]: Should they be a little more descriptive than "error"?

File: api/routes/Event.js [hunk 11]
```diff
@@ -14,11 +14,21 @@ const {
...
-    .then(items => res.status(OK).send(items));
+    .then(items => res.status(OK).send(items))
+    .catch(err => {
+      const info = {
+        errorTime: new Date(),
+        apiEndpoint: 'Event/getEvents',
+        errorDescription: 'error'
+      };
+      addErrorLog(info);
+      res.status(BAD_REQUEST).send({ err, message: 'deleting event failed' });
```
Comment (ThaiQ) [inline api/routes/Event.js:30]: @evanugarte do we still need to send the ```err message``` even though the message should be in the ```errorDescription```?

**Comment by evanugarte** [comment]
image

**Pull request merged**